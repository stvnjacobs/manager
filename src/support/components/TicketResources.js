import PropTypes from 'prop-types';
import React, { Component } from 'react';

import { Card, CardHeader } from 'linode-components';
import { ExternalLink } from 'linode-components';

export default class TicketResources extends Component {
  toggleSection(section) {
    return () =>
      this.setState({ hidden: { ...this.state.hidden, [section]: !this.state.hidden[section] } });
  }


  render() {
    const header = !this.props.displayHeader ? null : (
      <CardHeader title={'Resources'} />
    );

    const resources = [
      {
        link: 'https://forum.linode.com/',
        title: 'Community forum',
      },
      {
        link: 'https://linode.com/docs',
        title: 'User documentation and guides',
      },
      {
        link: 'https://developers.linode.com/',
        title: 'Deveoper documentation and guides',
      },
      {
        link: 'https://status.linode.com/',
        title: 'Overall system status',
      },
    ];

    const resourceItems = resources.map((resource) =>
      <li>
        <ExternalLink to={resource.link}>{resource.title}</ExternalLink>
      </li>
    );

    return (
      <Card
        header={header}
        className="TicketResources"
      >
        <ul className="list-unstyled">
          {resourceItems}
        </ul>
      </Card>
    );
  }
}

TicketResources.propTypes = {
  displayHeader: PropTypes.bool,
};
