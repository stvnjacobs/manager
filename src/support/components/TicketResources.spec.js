import React from 'react';
import { shallow } from 'enzyme';

import TicketResources from '~/support/components/TicketResources';

describe('support/components/TicketResources', () => {
  it('should render without error', () => {
    const wrapper = shallow(
      <TicketResources />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
