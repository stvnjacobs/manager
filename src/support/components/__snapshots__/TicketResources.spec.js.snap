// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`support/components/TicketResources should render without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TicketResources />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ul
        className="list-unstyled"
      >
        <li>
          <ExternalLink
            to="https://forum.linode.com/"
          >
            Community forum
          </ExternalLink>
        </li>
        <li>
          <ExternalLink
            to="https://linode.com/docs"
          >
            User documentation and guides
          </ExternalLink>
        </li>
        <li>
          <ExternalLink
            to="https://developers.linode.com/"
          >
            Deveoper documentation and guides
          </ExternalLink>
        </li>
        <li>
          <ExternalLink
            to="https://status.linode.com/"
          >
            Overall system status
          </ExternalLink>
        </li>
      </ul>,
      "className": "TicketResources",
      "header": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <ExternalLink
              to="https://forum.linode.com/"
            >
              Community forum
            </ExternalLink>
          </li>,
          <li>
            <ExternalLink
              to="https://linode.com/docs"
            >
              User documentation and guides
            </ExternalLink>
          </li>,
          <li>
            <ExternalLink
              to="https://developers.linode.com/"
            >
              Deveoper documentation and guides
            </ExternalLink>
          </li>,
          <li>
            <ExternalLink
              to="https://status.linode.com/"
            >
              Overall system status
            </ExternalLink>
          </li>,
        ],
        "className": "list-unstyled",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ExternalLink
              to="https://forum.linode.com/"
            >
              Community forum
            </ExternalLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Community forum",
              "to": "https://forum.linode.com/",
            },
            "ref": null,
            "rendered": "Community forum",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ExternalLink
              to="https://linode.com/docs"
            >
              User documentation and guides
            </ExternalLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "User documentation and guides",
              "to": "https://linode.com/docs",
            },
            "ref": null,
            "rendered": "User documentation and guides",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ExternalLink
              to="https://developers.linode.com/"
            >
              Deveoper documentation and guides
            </ExternalLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Deveoper documentation and guides",
              "to": "https://developers.linode.com/",
            },
            "ref": null,
            "rendered": "Deveoper documentation and guides",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ExternalLink
              to="https://status.linode.com/"
            >
              Overall system status
            </ExternalLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Overall system status",
              "to": "https://status.linode.com/",
            },
            "ref": null,
            "rendered": "Overall system status",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ul
          className="list-unstyled"
        >
          <li>
            <ExternalLink
              to="https://forum.linode.com/"
            >
              Community forum
            </ExternalLink>
          </li>
          <li>
            <ExternalLink
              to="https://linode.com/docs"
            >
              User documentation and guides
            </ExternalLink>
          </li>
          <li>
            <ExternalLink
              to="https://developers.linode.com/"
            >
              Deveoper documentation and guides
            </ExternalLink>
          </li>
          <li>
            <ExternalLink
              to="https://status.linode.com/"
            >
              Overall system status
            </ExternalLink>
          </li>
        </ul>,
        "className": "TicketResources",
        "header": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <ExternalLink
                to="https://forum.linode.com/"
              >
                Community forum
              </ExternalLink>
            </li>,
            <li>
              <ExternalLink
                to="https://linode.com/docs"
              >
                User documentation and guides
              </ExternalLink>
            </li>,
            <li>
              <ExternalLink
                to="https://developers.linode.com/"
              >
                Deveoper documentation and guides
              </ExternalLink>
            </li>,
            <li>
              <ExternalLink
                to="https://status.linode.com/"
              >
                Overall system status
              </ExternalLink>
            </li>,
          ],
          "className": "list-unstyled",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ExternalLink
                to="https://forum.linode.com/"
              >
                Community forum
              </ExternalLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Community forum",
                "to": "https://forum.linode.com/",
              },
              "ref": null,
              "rendered": "Community forum",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ExternalLink
                to="https://linode.com/docs"
              >
                User documentation and guides
              </ExternalLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "User documentation and guides",
                "to": "https://linode.com/docs",
              },
              "ref": null,
              "rendered": "User documentation and guides",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ExternalLink
                to="https://developers.linode.com/"
              >
                Deveoper documentation and guides
              </ExternalLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Deveoper documentation and guides",
                "to": "https://developers.linode.com/",
              },
              "ref": null,
              "rendered": "Deveoper documentation and guides",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ExternalLink
                to="https://status.linode.com/"
              >
                Overall system status
              </ExternalLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Overall system status",
                "to": "https://status.linode.com/",
              },
              "ref": null,
              "rendered": "Overall system status",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
